Implements #3069 - allows use of pre-hashed passwords with userpass backend (#24651)
hashicorp/vault: website/content/api-docs/auth/userpass.mdx

### Parameters

- `username` `(string: <required>)` – The username for the user. Accepted characters: alphanumeric plus "_", "-", "." (underscore, hyphen and period); username cannot begin with a hyphen, nor can it begin or end with a period.

> - `password` `(string: <required if password_hash is not given>)` - The password for the user. Only required when creating the user.

Reviewer: ```suggestion
- `password` `(string: <required if password_hash is not given>)` - Password for the current user. Only required when creating the user.  Mutually exclusive with `password_hash`.
```

updated to match the language in `password_hash`

> - `password_hash` `(string: <required if password is not given>)` - Pre-hashed version of this user's password in bcrypt format. Mutually exclusive with `password`.

Reviewer: ```suggestion
- `password_hash` `(string: <required if password is not given>)` - Pre-hashed password for the current user in bcrypt format. Mutually exclusive with `password`.
```
